### PR TITLE
Populate node status transition timestamp and reason

### PR DESCRIPTION
**Pull request by deyuanix**
Post node last transition time and status reason.

### CODE REVIEW

File: pkg/cloudprovider/controller/nodecontroller.go [hunk 1]
```diff
@@ -262,26 +263,46 @@ func (s *NodeController) DoChecks(nodes *api.NodeList) *api.NodeList {
...
 	var conditions []api.NodeCondition
+
+	// Check Condition: NodeReady. TODO: More node conditions.
+	oldReadyCondition := s.getCondition(node, api.NodeReady)
+	newReadyCondition := s.checkNodeReady(node)
+	if oldReadyCondition == nil {
+		newReadyCondition.LastTransitionTime = util.Now()
+	} else {
+		newReadyCondition.LastTransitionTime = oldReadyCondition.LastTransitionTime
+		if !reflect.DeepEqual(oldReadyCondition, newReadyCondition) {
```
Comment (bgrant0607) [inline pkg/cloudprovider/controller/nodecontroller.go]: Why do you need DeepEqual rather than just checking the status field?

Comment (ddysher) [inline pkg/cloudprovider/controller/nodecontroller.go]: Done

**Comment by ddysher** [comment]
PTAL

File: pkg/cloudprovider/controller/nodecontroller.go [hunk 2]
```diff
@@ -262,26 +263,42 @@ func (s *NodeController) DoChecks(nodes *api.NodeList) *api.NodeList {
 // DoCheck performs health checking for given node.
 func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 	var conditions []api.NodeCondition
+
+	// Check Condition: NodeReady. TODO: More node conditions.
+	oldReadyCondition := s.getCondition(node, api.NodeReady)
+	newReadyCondition := s.checkNodeReady(node)
+	newReadyCondition.LastTransitionTime = util.Now()
```
Comment (bgrant0607) [inline pkg/cloudprovider/controller/nodecontroller.go]: I hate to nitpick, but why not just put this in an else case?

Comment (ddysher) [inline pkg/cloudprovider/controller/nodecontroller.go]: I haven't gave it much thought.  Sometimes I do use this pattern, i.e. "The default value will be x, unless condition c is true, which we then set the value to y".  This doesn't work well here I guess.

**Comment by bgrant0607** [comment]
LGTM.

**Pull request merged**